src/fsspec_xrootd/xrootd.py

```diff
@@ -219,6 +321,163 @@ def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list[Any]:
             else:
                 return [os.path.basename(item["name"].rstrip("/")) for item in listing]
 
+    async def _cat_file(self, path: str, start: int, end: int, **kwargs: Any) -> Any:
+        _myFile = client.File()
+        try:
+            status, _n = await _async_wrap(
+                _myFile.open,
+                self.protocol + "://" + self.storage_options["hostid"] + "/" + path,
+                OpenFlags.READ,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"File failed to read: {status.message}")
+            status, data = await _async_wrap(
+                _myFile.read,
+                start,
+                end - start,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"Bytes failed to read from open file: {status.message}")
+            return data
+        except Exception as exc:
+            status, _n = await _async_wrap(
+                _myFile.close,
+                self.timeout,
+            )
+            raise OSError(f"Something went wrong in _cat_file(): {exc}")
+
+    async def _get_max_chunk_info(self, file: Any) -> tuple[int, int]:
+        data_server = file.get_property("DataServer")
+        if data_server not in XRootDFileSystem.cat_chunk_cache.keys():
+            status, result = await _async_wrap(
+                self._myclient.query, QueryCode.CONFIG, "readv_iov_max readv_ior_max"
+            )
+            if not status.ok:
+                raise OSError(
+                    f"Server query for vector read info failed: {status.message}"
+                )
+            max_num_chunks, max_chunk_size = map(int, result.split(b"\n", 1))
+            XRootDFileSystem.cat_chunk_cache.update(
+                {
+                    data_server: {
+                        "max_num_chunks": max_num_chunks,
+                        "max_chunk_size": max_chunk_size,
+                    }
+                }
+            )
+        return (
+            int(XRootDFileSystem.cat_chunk_cache[data_server]["max_num_chunks"]),
+            int(XRootDFileSystem.cat_chunk_cache[data_server]["max_chunk_size"]),
+        )
```

Comment:
```python
info = XRootDFileSystem.cat_chunk_cache[data_server]
return (info["max_num_chunks"], info["max_chunk_size"])
```
I guess `cat_chunk_cache` might better be named `_dataserver_info_cache`

Reply:
ok